### PR TITLE
92 - Staging environment support.

### PR DESCRIPTION
**Pull request by srggsf**
Allows to use sandbox environment to not fill production with test data.

### CODE REVIEW

**Comment by UweTrottmann** [comment]
Thanks! I hope to have a look soon.

Note to self: if used for tests, requires adding new credentials.
https://staging.trakt.tv/oauth/applications/new

File: src/main/java/com/uwetrottmann/trakt5/TraktV2.java [hunk 1]
```diff
@@ -44,14 +45,14 @@
...
     public static final String API_URL = "https://" + API_HOST + "/";
+    public static final String API_STAGING_URL = "https://" + API_STAGING_HOST + "/";
     public static final String API_VERSION = "2";
 
     public static final String SITE_URL = "https://trakt.tv";
     public static final String OAUTH2_AUTHORIZATION_URL = SITE_URL + "/oauth/authorize";
-    public static final String OAUTH2_DEVICE_CODE_URL = API_URL + "oauth/device/code";
-    public static final String OAUTH2_DEVICE_TOKEN_URL = API_URL + "oauth/device/token";
-    public static final String OAUTH2_TOKEN_URL = SITE_URL + "/oauth/token";
+    public static final String OAUTH2_TOKEN_URL = "oauth/token";
```
Comment (UweTrottmann) [inline src/main/java/com/uwetrottmann/trakt5/TraktV2.java:55]: That looks wrong, why was this changed?

File: src/main/java/com/uwetrottmann/trakt5/TraktV2Interceptor.java [hunk 2]
```diff
@@ -28,7 +28,7 @@ public Response intercept(Chain chain) throws IOException {
     public static Response handleIntercept(Chain chain, String apiKey,
             @Nullable String accessToken) throws IOException {
         Request request = chain.request();
-        if (!TraktV2.API_HOST.equals(request.url().host())) {
+        if (!(TraktV2.API_HOST.equals(request.url().host()) || TraktV2.API_STAGING_HOST.equals(request.url().host()))) {
```
Comment (UweTrottmann) [inline src/main/java/com/uwetrottmann/trakt5/TraktV2Interceptor.java:31]: Can this use `trakt.apiHost()` as well?

File: src/main/java/com/uwetrottmann/trakt5/services/Authentication.java [hunk 3]
```diff
@@ -39,7 +39,7 @@
      * The {@code user_code} and {@code verification_url} should be presented to the user.
      * @param clientId Application Client Id
      */
-    @POST(TraktV2.OAUTH2_DEVICE_CODE_URL)
+    @POST("oauth/device/code")
```
Comment (UweTrottmann) [inline src/main/java/com/uwetrottmann/trakt5/services/Authentication.java:42]: Why the change? Rather keep the public field to not break existing users.

**Comment by UweTrottmann** [comment]
Thanks again. I had some comments before this can be merged.

File: src/main/java/com/uwetrottmann/trakt5/TraktV2.java (hunk 1 above)
Comment (srggsf) [reply src/main/java/com/uwetrottmann/trakt5/TraktV2.java:55]: Relative paths are used in the rest of retrofit interfaces. That paths were exception. I tried to make it all in the same style.

File: src/main/java/com/uwetrottmann/trakt5/TraktV2Interceptor.java (hunk 2 above)
Comment (srggsf) [reply src/main/java/com/uwetrottmann/trakt5/TraktV2Interceptor.java:31]: this is static context. If you change signature and make it instance method, then it will be possible to get rid of TraktV2.API_HOST and TraktV2.API_STAGING_HOST constants.

File: src/main/java/com/uwetrottmann/trakt5/services/Authentication.java (hunk 3 above)
Comment (srggsf) [reply src/main/java/com/uwetrottmann/trakt5/services/Authentication.java:42]: I tried to make everything to be in the same style (relative paths for every retrofit interface)

**Comment by srggsf** [comment]
Hello, 
attempt #2 is ready :)
please review again.

Thank you.

File: src/main/java/com/uwetrottmann/trakt5/TraktV2Interceptor.java (hunk 2 above)
Comment (srggsf) [reply src/main/java/com/uwetrottmann/trakt5/TraktV2Interceptor.java:32]: I made 2 versions.

File: src/main/java/com/uwetrottmann/trakt5/TraktV2.java (hunk 1 above)
Comment (UweTrottmann) [reply src/main/java/com/uwetrottmann/trakt5/TraktV2.java:57]: Thanks for spotting this (already fixed separately, but forgot to say thanks here), it was wrong this whole time!

**Comment by UweTrottmann** [comment]
Merged manually, will finally ship with `6.9.0`. Thanks again!
